### PR TITLE
chore(tiering): Range functions + small refactoring

### PR DESCRIPTION
**Pull request by dranikpg**
* STRLEN had an important bug
* Add support for range functions
* Remove some unnecessary code

### CODE REVIEW

**Comment by dranikpg** [comment]
Suspiciously the mixed append test is failing, shouldn't be triggered by this code 🤨

**Comment by dranikpg** [comment]
Finally fixed a bug in this PR🤦🏻

File: src/server/string_family.cc [hunk 1]
```diff
@@ -61,81 +61,109 @@ string GetString(const PrimeValue& pv) {
...
-  if (range_len == 0) {
-    auto it_res = db_slice.FindReadOnly(op_args.db_cntx, key, OBJ_STRING);
-    if (it_res) {
-      return it_res.value()->second.Size();
-    } else {
-      return it_res.status();
-    }
+  // For external entries we have to enqueue reads because modify operations like append could be
+  // already pending.
+  // TODO: Optimize to return co.Size() if no modify operations are present
```
Comment (dranikpg) [inline src/server/string_family.cc:87]: so yes, this is a pain point

File: src/server/db_slice.cc [hunk 2]
```diff
@@ -1399,6 +1406,7 @@ void DbSlice::ClearEntriesOnFlush(absl::Span<const DbIndex> indices, const DbTab
 
     // Wait for delete operations to finish in sync
     while (!async && db_ptr->stats.tiered_entries > 0) {
+      VLOG(0) << db_ptr->stats.tiered_entries;
```
Comment (romange) [inline src/server/db_slice.cc]: remove?

Comment (dranikpg) [reply src/server/db_slice.cc]: removed in subsequent

**Review by romange — APPROVED** [review]

**Pull request merged**